conseil-common/src/main/scala/tech/cryptonomic/conseil/common/rpc/RpcClient.scala

```diff
@@ -0,0 +1,93 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.util.control.NoStackTrace
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import org.http4s.{EntityDecoder, EntityEncoder, Header, Method, Uri}
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+// JSON-RPC client according to specification: https://www.jsonrpc.org/specification
+class RpcClient[F[_]: Concurrent](
+    endpoint: String,
+    maxConcurrent: Int,
+    httpClient: Client[F],
+    headers: Header*
+) extends Http4sClientDsl[F]
+    with LazyLogging {
+
+  def stream[P, R](batchSize: Int)(request: Stream[F, RpcRequest[P]])(
+      implicit decode: EntityDecoder[F, Seq[RpcResponse[R]]],
+      encode: EntityEncoder[F, List[RpcRequest[P]]]
+  ): Stream[F, R] =
+    request
+      .chunkN(batchSize)
+      .evalTap { requests =>
+        Concurrent[F].delay(logger.debug(s"Call RPC in a batch of: ${requests.size}"))
+      }
+      .mapAsyncUnordered(maxConcurrent) { chunks =>
```

Comment:
According to the JSON-RPC spec:

> The Server MAY process a batch rpc call as a set of concurrent tasks, processing them in any order and with any width of parallelism.

so the order is irrelevant

Reply:
Ok, I see now why.

Yet repeating this information here, where it has impact, sounds useful still to an ignorant reader like me, that doesn't come fresh with the json-rpc spec.
Since the spec is far detached from this specific piece of code, I might be left to wonder. Much less probably so if a nice pointer like your quote was put here as a doc comment.
I would read the comment and immediately be reminded that I might need to read the spec to fully grasp this and other pieces of the client.